### PR TITLE
code refactoring

### PR DESCRIPTION
**Pull request by naazarchiik**
simple code refactoring. improvement of level choices.  Used readonly for _mementos and _boardFactory in TicTacToe class. Renamed the private field Checks to _checks to follow naming conventions in CheckWins. String builder added in Player class

### CODE REVIEW

**Comment by SofiiaKozlyk** [comment]
I can't run the project because there are errors.
IMG_20240528_121456_041

File: TicTacToeClassLibrary/TicTacToe.cs [hunk 1]
```diff
@@ -46,19 +41,30 @@ public void WriteSign(int position)
...
 
-        public Board Create()
+        public Board CreateBoard()
         {
-            Console.WriteLine("Please select a board type (3x3, 4x4)");
+            Console.WriteLine("Please select a board type (3x3, 4x4):");
             string boardType = Console.ReadLine();
-            Board board = _boardFactory.Create(boardType);
-            _mementos = new List<IMemento>();
-            return board;
```
Comment (SofiiaKozlyk) [inline TicTacToeClassLibrary/TicTacToe.cs:55]: There is no creation of a new _mementos list (or deletion of all its elements). So when a player presses "u" when starting a new game, the board from the previous game is returned - this is a bug.

File: TicTacToeClassLibrary/TicTacToe.cs [hunk 2]
```diff
@@ -46,19 +41,30 @@ public void WriteSign(int position)
...
+            return _boardFactory.Create(boardType);
+        }
```
Comment (SofiiaKozlyk) [inline TicTacToeClassLibrary/TicTacToe.cs]: There is no creation of a new _mementos list (or deletion of all its elements). So when a player presses "u" when starting a new game, the board from the previous game is returned - this is a bug.

**Comment by naazarchiik** [comment]
All comments have been taken into account and corrected)

**Pull request merged**